homeassistant/components/knx/switch.py

```diff
@@ -54,6 +54,22 @@ def __init__(self, xknx: XKNX, config: ConfigType) -> None:
             )
         )
         self._unique_id = f"{self._device.switch.group_address}"
+        self._total_energy_sensor = XknxSensor(
+            xknx,
+            name="Total Energy kWh",
+            group_address_state=config.get(
+                SwitchSchema.CONF_TOTAL_ENERGY_USAGE_STATE_ADDRESS
+            ),
+            value_type="active_energy_kwh",
+        )
```

Comment:
```suggestion
        self._total_energy_sensor: XknxSensor | None = None
        if _total_engergy_ga := config.get(SwitchSchema.CONF_TOTAL_ENERGY_USAGE_STATE_ADDRESS):
            self._total_energy_sensor = XknxSensor(
                xknx,
                name="Total Energy kWh",
                group_address_state=_total_engergy_ga,
                value_type="active_energy_kwh",
            )
```
We could omit creation of empty xknx devices when this function is not configured.
Then we can just check for `None` instead of `self._total_energy_sensor.sensor_value.readable` or `self._total_energy_sensor.sensor_value.initialized`.

Reply:
Done